app/requests/remote/role.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Client4} from '@client/rest';
+import Operator from '@database/operator';
+import {getRoles} from '@queries/role';
+import {IsolatedEntities} from '@typings/database/enums';
+import {getActiveServerDatabase} from '@utils/database';
+
+export const loadRolesIfNeeded = async (updatedRoles: string[]) => {
```

Comment:
```suggestion
export const fetchRolesIfNeeded = async (updatedRoles: string[]) => {
```

Reply:
Let's chat about having the requests only making network requests and then have functions for querying and functions for preparing/inserting/delete/update instead of doing all 3 in the one function